### ISSUE TITLE
Null resilient props should not get nullable initial state in constructors

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#77991]
**Version Used**: 5.0.0-1.25203.3 (f72da0bd)

**Steps to Reproduce**:

```cs
[Fact]
public void Nullable_Resilient_InitialStateInConstructor()
{
    var source = """
        #nullable enable

        public class C
        {
            public string Prop => field ??= "a";

            public C(bool ignored)
            {
                Prop.ToString(); // unexpected warning
            }

            public C() : this(false)
            {
                Prop.ToString(); // unexpected warning
            }

            public void M()
            {
                Prop.ToString();
            }
        }
        """;
    var comp = CreateCompilation(source);
    comp.VerifyEmitDiagnostics();
}
```

**Expected Behavior**: No warnings in the above test

**Actual Behavior**: `warning CS8602: Dereference of a possibly null reference.` reported on the indicated line

It's a minor thing, but, basically, we are not enforcing a nullable state on the way out of the ctor, we probably shouldn't setup a nullable state on the way in. Note that this also happens when the constructor has a `C() : this(...)` initializer.

This is mostly related to the system which makes the property and field share a single slot. There are likely edge cases to work out when field initializers are used or when the property is assigned. See also commit 66709555506b1a596fd3e91783fcb426697ae7f7. There are likely more bugs related to this slot sharing, e.g. `myStruct = default; myStruct.Prop.ToString();` missing a warning.

But, I was thinking that using a property in a constructor without initializing it is a rare enough case that we can punt on the increased precision here for 17.14, and focus on core scenarios and reliability.
